packages/server/worker/src/lib/flow-worker.ts

```diff
@@ -13,7 +14,45 @@ let workerToken: string
 let heartbeatInterval: NodeJS.Timeout
 let socket: Socket
 
-export const flowWorker = (log: FastifyBaseLogger) => ({
+export const workerSocket = (log: FastifyBaseLogger) => ({
+    emitWithAck: async (event: string, data: unknown, options?: { timeoutMs?: number, retries?: number, retryDelayMs?: number }): Promise<void> => {
+        const timeoutMs = options?.timeoutMs ?? 10000
+        const retries = options?.retries ?? 3
+        const retryDelayMs = options?.retryDelayMs ?? 2000
+
+        for (let attempt = 0; attempt <= retries; attempt++) {
+            try {
+                if (!socket || !socket.connected) {
+                    throw new Error('Socket not connected')
+                }
+                await socket.timeout(timeoutMs).emitWithAck(event, data)
```

Comment:
Remove all these retries, and use https://socket.io/docs/v4/tutorial/step-8#:~:text=or%20with%20the%20retries%20option%3A